### PR TITLE
feat: Get, Patch, Delete /lobby/:id routes

### LINKED ISSUE #25: 🚀 Feature: PATCH /lobby/:id
[Design doc description](https://github.com/UCLA-Creative-Labs/aurgy-backend/blob/main/docs/DESIGN_DOC.md#PATCH-lobbyid)

Updates the lobby name

Verification happens in two stages:
1. User verification: make sure the user is valid
2. Lobby Permissions: Make sure the user is the manager of the lobby

Also maybe add lobby name sanitizing

* [ ] :wave: I may be able to implement this feature request

---

This is a :rocket: Feature Request

### LINKED ISSUE #26: 🚀 Feature: GET /lobby/:id
[Design doc description](https://github.com/UCLA-Creative-Labs/aurgy-backend/blob/main/docs/DESIGN_DOC.md#GET-lobbyid)

Get lobby info (name, theme, list of song ids, participants, playlist ID(?), manager ID(?))

* [ ] :wave: I may be able to implement this feature request

---

This is a :rocket: Feature Request

### LINKED ISSUE #30: 🚀 Feature: DELETE /lobby/:id/user/:id
The DELETE request for /lobby/:id/user/:id is a way for the manager of a lobby to remove another user from the lobby.

### Proposed Solution
- Make sure the User is valid
- Verify User is the manager of the lobby
- Remove User from Lobby participant's list
- Remove Lobby from User's participatingLobbies list

* [ ] :wave: I may be able to implement this feature request

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by trevorong**
- implement Get, Patch, Delete /lobby/:id routes
- fix some lobby creation bugs + add removeUser checks

closes #25, closes #26, closes #30

### CODE REVIEW

File: www/lobby/:id.ts [hunk 1]
```diff
@@ -15,15 +17,39 @@ lobby_id_router.post('/:id', async (req: Request, res: Response) => {
...
  */
 lobby_id_router.get('/:id', async (req: Request, res: Response) => {
-  res.status(200).json('placeholder get');
+  const lobbyId = req.params.id;
+  const userId = req.body.id;
+  const lobby = await Lobby.fromId(lobbyId);
+  if (!lobby) return res.status(404).json('Lobby not found in database').end();
+
+  const user = await User.fromId(userId);
+  if (!user) return res.status(404).json('User not found in database').end();
```
Comment (trevorong) [inline www/lobby/:id.ts:26]: this is duplicated for each of the routes b/c we need to make sure the lobbyId/userIds are valid and get the objects from the ids but I'm not sure what the best way to get rid of this is

Comment (trevorong) [reply www/lobby/:id.ts:26]: using middleware would work but I don't think you can pass objects between handlers so each handler would still  need to get User.fromId and Lobby.fromId

Comment (BryanPan342) [reply www/lobby/:id.ts:26]: How about we just make it a function?

File: lib/lobby.ts [hunk 2]
```diff
@@ -65,8 +65,13 @@ export class Lobby extends DbItem implements ILobby {
    * @returns a newly created Lobby object
    */
   public static async create(props: LobbyCreateProps, key : string | null = null) : Promise<Lobby | null> {
-    const playlistId = await createSpotifyPlaylist();
-    const participants = [props.managerId];
+    const managerId = props.managerId;
+    const playlistId = await createSpotifyPlaylist(managerId);
```
Comment (BryanPan342) [inline lib/lobby.ts:69]: Why are we passing the managerId?

File: lib/lobby.ts [hunk 3]
```diff
@@ -65,8 +65,13 @@ export class Lobby extends DbItem implements ILobby {
...
+    if (!playlistId) return null;
+
+    const exists = await Lobby.fromId(playlistId);
```
Comment (BryanPan342) [inline lib/lobby.ts:72]: I doubt this will be possible bc Spotify will return a unique id

**Review by BryanPan342 — COMMENTED** [review]
General first pass comments

File: www/lobby/:id.ts [hunk 4]
```diff
@@ -15,15 +17,35 @@ lobby_id_router.post('/:id', async (req: Request, res: Response) => {
...
 
 // Update lobby information
 /**
  * Body Params: id, name, refreshToken
  */
 lobby_id_router.patch('/:id', async (req: Request, res: Response) => {
-  res.status(200).json('placeholder patch');
+  const lobbyId = req.params.id;
+  const userId = req.body.id;
+  const name : string = req.body.lobbyName;
```
Comment (BryanPan342) [inline www/lobby/:id.ts:37]: nit
```suggestion
  const name: string = req.body.lobbyName;
```

**Review by BryanPan342 — COMMENTED** [review]
small nit, otherwise everything else looks good!

**Review by BryanPan342 — APPROVED** [review]
LGTM 🥳🥳

**Pull request merged**